layouts/community/ortho_4x12/junonum/keymap.c

```diff
@@ -0,0 +1,346 @@
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        print("mode just switched to qwerty and this is a huge string\n");
+        set_single_persistent_default_layer(_QWERTY);
+      }
+      return false;
+      break;
+    case COLEMAK:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_COLEMAK);
+      }
+      return false;
+      break;
+    case DVORAK:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_DVORAK);
+      }
+      return false;
+      break;
+    case BACKLIT:
+      if (record->event.pressed) {
+        register_code(KC_RSFT);
+        #ifdef BACKLIGHT_ENABLE
+          backlight_step();
+        #endif
+        #ifdef KEYBOARD_planck_rev5
+          PORTE &= ~(1<<6);
+        #endif
+      } else {
+        unregister_code(KC_RSFT);
+        #ifdef KEYBOARD_planck_rev5
+          PORTE |= (1<<6);
+        #endif
+      }
+      return false;
+      break;
+    case PLOVER:
+      if (record->event.pressed) {
+        #ifdef AUDIO_ENABLE
+          stop_all_notes();
+          PLAY_SONG(plover_song);
+        #endif
+        layer_off(_RAISE);
+        layer_off(_LOWER);
+        layer_off(_ADJUST);
+        layer_on(_PLOVER);
+        if (!eeconfig_is_enabled()) {
+            eeconfig_init();
+        }
+        keymap_config.raw = eeconfig_read_keymap();
+        keymap_config.nkro = 1;
+        eeconfig_update_keymap(keymap_config.raw);
+      }
+      return false;
+      break;
+    case EXT_PLV:
+      if (record->event.pressed) {
+        #ifdef AUDIO_ENABLE
+          PLAY_SONG(plover_gb_song);
+        #endif
+        layer_off(_PLOVER);
+      }
+      return false;
+      break;
+	  
+	  
+// Play audio upon switching Caps Lock and custom layers
+
+	case KC_CAPS:
+		if (record->event.pressed) {
+			#ifdef AUDIO_ENABLE
+			if (host_keyboard_leds() & (1<<USB_LED_CAPS_LOCK)) {
+				PLAY_SONG(caps_song_off);
+			} else {
+				PLAY_SONG(caps_song_on);
+			}
+			#endif
+		}
+		return true;
+		
+	case DP_ON:
+		if (record->event.pressed) {
+			
+		} else { // only actived upon key release
+			#ifdef AUDIO_ENABLE
+			PLAY_SONG(dpad_song_on);
+			#endif
+			
+			layer_off(_FN1);
+			layer_on(_DPAD);
+		}
+		
+	case DP_OFF:
+		if (record->event.pressed) {
+			#ifdef AUDIO_ENABLE	
+			PLAY_SONG(dpad_song_off);
+			#endif
+			
+			layer_off(_DPAD);	
+		}
+    }
+  return true;
+}
+
+bool muse_mode = false;
+uint8_t last_muse_note = 0;
+uint16_t muse_counter = 0;
+uint8_t muse_offset = 70;
+uint16_t muse_tempo = 50;
+
+void encoder_update(bool clockwise) {
+  if (muse_mode) {
+    if (IS_LAYER_ON(_RAISE)) {
+      if (clockwise) {
+        muse_offset++;
+      } else {
+        muse_offset--;
+      }
+    } else {
+      if (clockwise) {
+        muse_tempo+=1;
+      } else {
+        muse_tempo-=1;
+      }
+    }
+  } else {
+    if (clockwise) {
+      register_code(KC_PGDN);
+      unregister_code(KC_PGDN);
```

Comment:
These two can be merged into simply `tap_code(KC_PDGN)`, and the same for the `KC_PGUP`s below.